com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -366,53 +348,112 @@ internal Vector3 Scale
         /// </summary>
         protected NetworkManager m_CachedNetworkManager;
 
-        private readonly NetworkVariable<NetworkTransformState> m_ReplicatedNetworkState = new NetworkVariable<NetworkTransformState>(new NetworkTransformState());
+        /// <summary>
+        /// Used to know when we are resetting the dirty flag (i.e. we sent the state)
+        /// </summary>
+        internal class NetworkTransformVariable : NetworkVariable<NetworkTransformState>
+        {
+            public override void ResetDirty()
+            {
+                base.ResetDirty();
 
-        private NetworkTransformState m_LocalAuthoritativeNetworkState;
+                // Now clear our bitset and prepare for next network tick state update
+                Value.ClearBitSetForNextTick();
+            }
+
+            public NetworkTransformVariable(NetworkTransformState value = default,
+                NetworkVariableReadPermission readPerm = DefaultReadPerm,
+                NetworkVariableWritePermission writePerm = DefaultWritePerm)
+                : base(value, readPerm, writePerm)
+            {
+            }
+        }
+
+        /// <summary>
+        /// We have two internal NetworkVariables.
+        /// One for server authoritative and one for "client/owner" authoritative.
+        /// </summary>
+        private readonly NetworkTransformVariable m_ReplicatedNetworkStateServer = new NetworkTransformVariable(new NetworkTransformState(), NetworkVariableReadPermission.Everyone, NetworkVariableWritePermission.Server);
+        private readonly NetworkTransformVariable m_ReplicatedNetworkStateOwner = new NetworkTransformVariable(new NetworkTransformState(), NetworkVariableReadPermission.Everyone, NetworkVariableWritePermission.Owner);
```

Comment:
Since owner driven transform will be done by overriding NetworkTransform, shouldn't this be done by users when they override it? See PR here https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/1873

Reply:
We need to define two NetworkVariables as they have to be instantiated when declared (last I checked) in order to work properly.

Reply:
My comment above with user defined ClientNetworkTransform would still work, as the child class would declare it the same way. I guess yours works as well, since you have that wrapper netvar that abstracts it.